MapboxVision/src/main/java/com/mapbox/vision/telemetry/TotalBytesCounter.kt

```diff
@@ -12,45 +16,51 @@ internal interface TotalBytesCounter {
 
     fun millisToNextSession(): Long
 
+    fun fitInLimit(bytes: Long): Boolean
+
     class Impl(
-            private val sessionLengthMillis: Long = SESSION_LENGTH_MILLIS,
-            private val sessionMaxBytes: Long = SESSION_MAX_BYTES
+        private val sessionLengthMillis: Long = SESSION_LENGTH_MILLIS,
+        private val sessionMaxBytes: Long = SESSION_MAX_BYTES,
+        private val systemTime: SystemTime = SystemTime.Impl
     ) : TotalBytesCounter {
 
         companion object {
-            private const val SESSION_LENGTH_MILLIS = 60 * 60 * 1000L // one hour
+            private val SESSION_LENGTH_MILLIS by lazy(LazyThreadSafetyMode.NONE) { TimeUnit.HOURS.toMillis(1) }
             private const val SESSION_MAX_BYTES = 30 * 1024 * 1024L // 30 MB
         }
 
         private var sessionStartMillis: Long = 0
         private var bytesSentPerSession: Long = 0
 
         override fun trackSentBytes(bytes: Long): Boolean {
-
-            val timestamp = System.currentTimeMillis()
+            val timestamp = systemTime.currentTimeMillis()
 
             return when {
-                sessionStartMillis + sessionLengthMillis < timestamp -> {
+                sessionStartMillis + sessionLengthMillis <= timestamp -> {
                     sessionStartMillis = timestamp
-                    bytesSentPerSession = bytes
-                    true
+                    bytesSentPerSession = 0
+                    return if (fitInLimit(bytes)) {
```

Comment:
Good catch! Should we add ticket to handle such situations?

Reply:
Yep, I think so. A delgate/listener should notify user about it

Reply:
Do you plan to add it in this PR? Or create separate ticket to handle this case?

Reply:
I think this PR is about small refactoring, setup test environment and test introducing. I'd like to create separate ticket. I'll make it.

Reply:
Sure, great!